### DISCUSSION TITLE
Implement DNS in Archer C7 tplink

### DISCUSSION THREAD

**Discussion by pmusetti** [Ultimate-Hosts-Blacklist/Ultimate.Hosts.Blacklist#689] [Q&A]
Hello, I want to implement this DNS in my archer C7. Can anybody tell me how to proceed?

**Comment by spirillen** [comment]
What is archer C7??

**Comment by Somebodyisnobody** [comment]
It's an TP-Link router model.

@pmusetti As far as i know the Archer C7 is a good paperweight regarding software features. I recommend you to use a pihole (Raspberry pi with this software here: https://pi-hole.net/) and change the archer's DNS settings to point to the pihole.

**Comment by pmusetti** [comment]
Ok, thanks.
I will try with pi-hole.
I set de dns in the tp-link and I still see ads

Pablo Musetti

**Reply by Somebodyisnobody** [reply]
Look into the pihole-logs if it is used and gets requests. Did you add the ultimate hosts blacklist to the pihole? Not all ads are blocked (f.x. if the domain that offers ads also offers the website itself).

**Comment by smed79** [comment]
> I want to implement this DNS in my archer C7. Can anybody tell me how to proceed?

see https://www.tp-link.com/en/support/faq/1712/

> I set de dns in the tp-link and I still see ads

Provide an example page so that we can check.
